tests/forms_tests/tests/test_forms.py

```diff
@@ -2486,6 +2486,14 @@ class FileForm(forms.Form):
         self.assertEqual(f.errors, {})
         self.assertEqual(f.cleaned_data['file1'], 'resume.txt')
 
+    def test_fileinput_use_required_attribute(self):
+        """#31118 - remove `required` attribute if initial data exists"""
+        class FileForm(Form):
+            file1 = forms.FileField(initial=lambda: 'resume.txt', widget=FileInput)
+
+        f = FileForm()
+        self.assertHTMLEqual(str(f['file1']), """<input type="file" name="file1" id="id_file1">""")
+
```

Comment:
Please add a direct test in the `forms_tests/widget_tests/test_fileinput.py` (see [similar tests](https://github.com/django/django/commit/fab46ce6f5a0a58c4e5e39c9e5e412702beb4a64#diff-08f28eef028de035a3196b6c312a929fR147-R151)).